python/hail/expr/aggregators.py

```diff
@@ -481,12 +498,9 @@ def sum(expr):
     :class:`.Expression` of type :py:data:`.tint64` or :py:data:`.tfloat64`
         Sum of records of `expr`.
     """
-    agg = _to_agg(expr)
-    if not is_numeric(agg._type):
-        raise TypeError("'sum' expects a numeric argument, found '{}'".format(agg._type))
-    return _agg_func('sum', agg, agg._type)
+    return _agg_func('sum', expr, expr.dtype)
 
-@typecheck(expr=oneof(Aggregable, expr_any))
+@typecheck(expr=agg_expr(expr_array(expr_oneof(expr_int64, expr_float64))))
```

Comment:
seems like this should be expr_numeric, it looks int32 and float32 are implemented in the FR.

Reply:
will fix.